lib/logstash/outputs/rabbitmq.rb

```diff
@@ -76,20 +71,108 @@ def initialize(params)
     super
   end
 
-  # Use MarchHare on JRuby to avoid IO#select CPU spikes
-  # (see github.com/ruby-amqp/bunny/issues/95).
-  #
-  # On MRI, use Bunny.
-  #
-  # See http://rubybunny.info and http://rubymarchhare.info
-  # for the docs.
-  if RUBY_ENGINE == "jruby"
-    require "logstash/outputs/rabbitmq/march_hare"
+  public
```

Comment:
In my experience and after reading a lot of gem sources, this is my take...
```
requires
class
class methods if any (also attr_reader, def_delegator etc)
initialize
public methods (ordered top down in execution order, fluent ones upper most)
protected methods (ordered top down relative to their use in public methods)
private methods (ordered top down relative to their use in public methods)
```

AFAIK Ruby has a private keyword and a private method, the keyword form switches the parser into a define-methods-private state but the method form `private(:register)` alters the visibility of a previously defined method.  So if it is imperative that a private method is v near a public one we can do
```ruby
def public_method
  ...
  private_method(@cfg)
  ...
end

def private_method(cfg)

end
private(:private_method)
```